### ISSUE TITLE
Exception `DatasetInfo` object has no attribute `intra_dataset_path`

### ISSUE THREAD

**Issue by christian-monch** [datalad/datalad-metalad#296]
With the standard `meta-extract` conduct pipeline definition and a dataset with one or more subdatasets, the following command

```
> datalad meta-conduct extract_metadata traverser.top_level_dir=$(pwd) traverser.item_type=dataset traverser.traverse_sub_datasets=True extractor.extractor_type=dataset extractor.extractor_name=metalad_core
``` 

will result in an exception:
```
[ERROR  ] Exception 'DatasetInfo' object has no attribute 'intra_dataset_path' in processor <Future at 0x7f826d60f460 state=finished raised AttributeError> 
meta_conduct(error): [concurrent.futures.process._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/lib64/python3.10/concurrent/futures/process.py", line 246, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/home/cristian/Develop/datalad-metalad/datalad_metalad/pipeline/processor/base.py", line 25, in execute
    return context, self.process(pipeline_data)
  File "/home/cristian/Develop/datalad-metalad/datalad_metalad/pipeline/processor/extract.py", line 92, in process
    intra_dataset_path = dataset_traverse_record.element_info.intra_dataset_path
AttributeError: 'DatasetInfo' object has no attribute 'intra_dataset_path'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/cristian/Develop/datalad-metalad/datalad_metalad/conduct.py", line 376, in process_parallel
    source_index, pipeline_data = future.result()
  File "/usr/lib64/python3.10/concurrent/futures/_base.py", line 451, in result
    return self.__get_result()
  File "/usr/lib64/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
AttributeError: 'DatasetInfo' object has no attribute 'intra_dataset_path'

**Comment by christian-monch** [comment]
Cannot be reproduced on master or maint_0.4. Closing it.

**Comment by christian-monch** [comment]
Error appears in branch `enh-reduce-pipeline-processes`

**Comment by christian-monch** [comment]
Fixed in 578ebb348d18df1426c4a5def1cbe30ebef983a3
